feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/presentation/cloudBackup/BackupSettingsViewModel.kt

```diff
@@ -147,6 +187,52 @@ class CloudBackupSettingsViewModel(
 
         isSyncing.value = false
     }
+
+    private fun manageBackupChangePasswordItem(): ListSelectorMixin.Item {
+        return ListSelectorMixin.Item(
+            R.drawable.ic_pin,
+            R.color.icon_primary,
+            R.string.common_change_password,
+            R.color.text_primary,
+            ::onChangePasswordClicked
+        )
+    }
+
+    private fun manageBackupDeleteBackupItem(): ListSelectorMixin.Item {
+        return ListSelectorMixin.Item(
+            R.drawable.ic_delete,
+            R.color.icon_negative,
+            R.string.backup_settings_delete_backup,
+            R.color.text_negative,
+            ::onDeleteBackupClicked
+        )
+    }
+
+    private fun onChangePasswordClicked() {
+        TODO()
+    }
+
+    private fun onDeleteBackupClicked() {
+        actionBottomSheetLauncher.launchDeleteBackupAction(resourceManager, ::confirmCloudBackupDelete)
+    }
+
+    private fun confirmCloudBackupDelete() {
+        launch {
+            confirmationAwaitableAction.awaitDeleteBackupConfirmation()
+
+            progressDialogMixin.startProgress(R.string.deleting_backup_progress) {
+                cloudBackupSettingsInteractor.deleteCloudBackup()
+                    .onSuccess {
+                        cloudBackupSettingsInteractor.setCloudBackupSyncEnabled(false)
+                        cloudBackupEnabled.value = false
+                    }
+                    .onFailure { throwable ->
+                        val titleAndMessage = mapDeleteBackupFailureToUi(resourceManager, throwable)
+                        titleAndMessage?.let { showError(it) }
+                    }
```

Comment:
I suggest to move it to a separate function since there are many nested levels and it makes code harder to read